dc-cudami-editor/src/components/IdentifierSearch.jsx

```diff
@@ -11,113 +12,104 @@ import {
 
 import {findByIdentifier, loadIdentifiable} from '../api'
 import AppContext from './AppContext'
+import FeedbackMessage from './FeedbackMessage'
 import InputWithSpinner from './InputWithSpinner'
 import PreviewImage from './PreviewImage'
 import {getLabelValue} from './utils'
 
-class IdentifierSearch extends Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      id: '',
-      loading: false,
-    }
-  }
-
-  componentDidUpdate(prevProps) {
-    if (prevProps.namespace !== this.props.namespace) {
-      this.setState({
-        id: '',
-        loading: false,
-        result: undefined,
-      })
-    }
-  }
-
-  componentWillUnmount() {
-    this.props.setFeedbackMessage(undefined)
-  }
-
-  search = async () => {
-    this.setState({loading: true})
-    const {fixedIdentifiers, namespace, setFeedbackMessage, type} = this.props
-    const isFixedIdentifier = fixedIdentifiers.some(
-      (i) => i.namespace === namespace,
-    )
-    const result = await (isFixedIdentifier
-      ? loadIdentifiable(this.context.apiContextPath, type, this.state.id)
-      : findByIdentifier(
-          this.context.apiContextPath,
-          this.state.id,
-          namespace,
-          type,
-        ))
-    const isEmptyResult = Object.keys(result).length === 0
-    if (isEmptyResult) {
-      setFeedbackMessage({
-        color: 'warning',
-        key: `${type}NotFound`,
-      })
-    } else {
-      setFeedbackMessage(undefined)
-    }
-    this.setState({isEmptyResult, loading: false, result})
-  }
+const search = async (
+  {fixedIdentifiers, namespace, type},
+  apiContextPath,
+  id,
+) => {
+  const isFixedIdentifier = fixedIdentifiers.some(
+    (i) => i.namespace === namespace,
+  )
+  const suggestion = await (isFixedIdentifier
+    ? loadIdentifiable(apiContextPath, type, id)
+    : findByIdentifier(apiContextPath, id, namespace, type))
+  return suggestion
+}
 
-  render() {
-    const {id, isEmptyResult, loading, result} = this.state
-    return (
-      <>
-        <InputWithSpinner
-          inputProps={{
-            onChange: (evt) => this.setState({id: evt.target.value}),
-            value: id,
-          }}
-          loading={loading}
-        >
-          <InputGroupAddon addonType="append">
-            <Button
-              className="align-items-center d-flex"
-              color="primary"
-              disabled={!id.length}
-              onClick={this.search}
-            >
-              <FaSearch />
-            </Button>
-          </InputGroupAddon>
-        </InputWithSpinner>
-        {result && !isEmptyResult && (
-          <ListGroup className="suggestion-container">
+export const IdentifierSearch = (props) => {
+  const {activeLanguage, namespace, onSelect, type} = props
```

Comment:
Spricht was dagegen das Destructuring direkt in der Parameter-Definition zu machen?

Reply:
Die werden als Gesamtes an die `search`-Funktion übergeben, war so etwas kürzer :thinking: